src/status_im/extensions/core.cljs

```diff
@@ -1,7 +1,13 @@
 (ns status-im.extensions.core
-  (:require [clojure.string :as string]))
+  (:require [clojure.string :as string]
+            [status-im.utils.fx :as fx]))
 
 (defn url->storage-details [s]
   (when s
     (let [[_ type id] (string/split s #".*[:/]([a-z]*)@(.*)")]
       [(keyword type) id])))
+
+(fx/defn set-extension-url-from-qr
+  [{:keys [db]} url]
+  {:db       (assoc db :extension-url url)
+   :dispatch [:navigate-back]})
```

Comment:
Unnecessary dispatch, should be just `navigate-back` function from navigation ns.

Reply:
I don't agree.

`:navigate-back` is the only handler we use in this case, and it also has a check for modal dialog. The function you're talking about is a side-effect, it's a strange idea to just call it like that.

Reply:
it does the same as :navigate-back does without an unnecessary dispatch there is no side-effect it updates the db and adds the ::navigate-back nil effect

Reply:
```
(fx/defn navigate-back
  [{{:keys [navigation-stack view-id] :as db} :db}]
  (assoc
   {::navigate-back nil}
   :db (let [[previous-view-id :as navigation-stack'] (pop navigation-stack)
             first-in-stack (first navigation-stack)]
         (if (= view-id first-in-stack)
           (-> db
               (assoc :view-id previous-view-id)
               (assoc :navigation-stack navigation-stack'))
           (assoc db :view-id first-in-stack)))))

(handlers/register-handler-fx
 :navigate-back
 (re-frame/enrich -preload-data!)
 (fn [cofx _]
   (navigate-back cofx)))
```

Reply:
Then sorry, I simply looked at the wrong function